ui/tests/acceptance/secrets/backend/gcp/gcp-configuration-test.js

```diff
@@ -20,91 +21,235 @@ import {
   expectedConfigKeys,
   expectedValueOfConfigKeys,
   configUrl,
+  fillInGcpConfig,
 } from 'vault/tests/helpers/secret-engine/secret-engine-helpers';
 
 module('Acceptance | GCP | configuration', function (hooks) {
   setupApplicationTest(hooks);
   setupMirage(hooks);
 
   hooks.beforeEach(function () {
+    const flash = this.owner.lookup('service:flash-messages');
     this.store = this.owner.lookup('service:store');
+    this.flashSuccessSpy = spy(flash, 'success');
+    this.flashDangerSpy = spy(flash, 'danger');
+    this.flashInfoSpy = spy(flash, 'info');
     this.version = this.owner.lookup('service:version');
     this.uid = uuidv4();
     this.type = 'gcp';
+    this.path = `GCP-${this.uid}`;
     return authPage.login();
   });
-  module('isEnterprise', function (hooks) {
+
+  test('it should prompt configuration after mounting the GCP engine', async function (assert) {
+    await visit('/vault/settings/mount-secret-backend');
+    await mountBackend(this.type, this.path);
+
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${this.path}/configuration`,
+      'navigated to configuration view'
+    );
+    assert.dom(GENERAL.emptyStateTitle).hasText('Google Cloud not configured');
+    assert.dom(GENERAL.emptyStateActions).hasText('Configure Google Cloud');
+    // cleanup
+    await runCmd(`delete sys/mounts/${this.path}`);
+  });
+
+  test('it should transition to configure page on click "Configure" from toolbar', async function (assert) {
+    await enablePage.enable(this.type, this.path);
+    await click(SES.configure);
+    assert.strictEqual(
+      currentURL(),
+      `/vault/secrets/${this.path}/configuration/edit`,
+      'navigated to configuration edit view'
+    );
+    // cleanup
+    await runCmd(`delete sys/mounts/${this.path}`);
+  });
+
+  module('isCommunity', function (hooks) {
     hooks.beforeEach(function () {
-      this.version.type = 'enterprise';
+      this.version.type = 'community';
     });
 
-    test('it should show empty state and navigate to configuration view after mounting the GCP engine', async function (assert) {
-      const path = `GCP-${this.uid}`;
-      await visit('/vault/settings/mount-secret-backend');
-      await mountBackend(this.type, path);
-
-      assert.strictEqual(
-        currentURL(),
-        `/vault/secrets/${path}/configuration`,
-        'navigated to configuration view'
-      );
-      assert.dom(GENERAL.emptyStateTitle).hasText('Google Cloud not configured');
-      assert.dom(GENERAL.emptyStateActions).doesNotContainText('Configure GCP');
-      // cleanup
-      await runCmd(`delete sys/mounts/${path}`);
+    module('details', function () {
```

Comment:
do we need to wrap each of these tests in an additional module? It seems like slightly unnecessary nesting since each only contains one or two tests

Reply:
it helped me write the tests to make sure I covered the logic iterated over if the model was new or not. And details is a totally different route. While there's are only several tests in each, I think it helps break up the relevant logic.

Reply:
No need to change for this, I just think the modules should be used sparingly and for setting up shared test context. A lot of nesting seems like it's just setting us up for future flakiness